### PR TITLE
Checking missing parameters definitions in classes methods

### PR DESCRIPTION
**Pull request by YurkinPark**
Code checks missing parameters in classes methods. Ref https://www.drupal.org/node/2624558

### CODE REVIEW

**Comment by klausi** [comment]
Tests are failing because of coding standard errors, can you fix those?

File: coder_sniffer/Drupal/Test/bad/bad.php [hunk 1]
```diff
@@ -830,4 +830,24 @@ function test28() {
 
 }
 
+/**
+ * Class Test.
+ */
+class Test extends AbstractTest {
```
Comment (klausi) [inline coder_sniffer/Drupal/Test/bad/bad.php]: please don't add to bad.php. Can you move the test to the test file for this sniff?

File: coder_sniffer/Drupal/Sniffs/Commenting/FunctionCommentSniff.php [hunk 2]
```diff
@@ -806,6 +805,17 @@ protected function processParams(PHP_CodeSniffer_File $phpcsFile, $stackPtr, $co
             }
         }//end foreach
 
+        if ($tokens[$stackPtr]['level'] > 0) {
+            foreach ($realParams as $realParam) {
+                $realParamKeyName = $realParam['pass_by_reference'] ? '&' : '';
+                $realParamKeyName .= $realParam['name'];
+                if (!in_array($realParamKeyName, $foundParams)) {
+                    $error = 'Parameter %s is not described in comment';
+                    $phpcsFile->addError($error, $commentStart, 'Param', [$realParam['name']]);
```
Comment (klausi) [inline coder_sniffer/Drupal/Sniffs/Commenting/FunctionCommentSniff.php]: the third parameter should be more specific. "ParamMissing" maybe?

File: coder_sniffer/Drupal/Test/bad/bad.php (hunk 1 above)
Comment (YurkinPark) [inline coder_sniffer/Drupal/Test/bad/bad.php]: Test moved

File: coder_sniffer/Drupal/Sniffs/Commenting/FunctionCommentSniff.php (hunk 2 above)
Comment (YurkinPark) [inline coder_sniffer/Drupal/Sniffs/Commenting/FunctionCommentSniff.php]: Renamed to ParamMissingDefinition

File: coder_sniffer/Drupal/Sniffs/Commenting/FunctionCommentSniff.php [hunk 3]
```diff
@@ -806,6 +804,22 @@ protected function processParams(PHP_CodeSniffer_File $phpcsFile, $stackPtr, $co
             }
         }//end foreach
 
+        if ($tokens[$stackPtr]['level'] > 0) {
```
Comment (klausi) [inline coder_sniffer/Drupal/Sniffs/Commenting/FunctionCommentSniff.php]: why do we need this level check? Please add a comment.

Comment (klausi) [inline coder_sniffer/Drupal/Sniffs/Commenting/FunctionCommentSniff.php]: ah, it looks like you don't want to run this sniff on normal functions? But the param rules also apply to functions outside of classes, let's add a test case for that as well.

Comment (YurkinPark) [inline coder_sniffer/Drupal/Sniffs/Commenting/FunctionCommentSniff.php]: we have function which defined as hooks (there is standard comment "Implements hook_<hook_name>"

Comment (YurkinPark) [inline coder_sniffer/Drupal/Sniffs/Commenting/FunctionCommentSniff.php]: coder/coder_sniffer/Drupal/Test/Commenting/FunctionCommentUnitTest.1.inc

**Comment by YurkinPark** [comment]
I implemented this test and fixed code a bit (to let it pass this test). Review my last commit please.

**Comment by YurkinPark** [comment]
I just realized, i didn't cover this case:
```php
/**
 * {@inheritdoc}
 */
```

**Comment by klausi** [comment]
No worries, I committed a follow-up to deal with that :-)

**Comment by andypost** [comment]
For ref https://github.com/klausi/coder/commit/4756f958a56f9100a2412360e1d8455f2b93ad1c

**Pull request merged**